Automattic/wp-calypso
# Masterbar: clean up, component refactor, and move to gridicons

This issue is a deep refactoring of the Masterbar component.

Summary:

* Modular, normalized the components to just `masterbar-item*` and derivates.
* Restructured the DOM tree from 7-levels to 2-levels.
* Rewritten the SCSS almost entirely, using `flex` magic.
* Switched to ES6 syntax.
* Ready for logged-out / SSR masterbar work.
* Switches to Gridicons.

To test:

* Do a fresh load (cache off / cleared)
* Navigate to the top five sections (My Sites, Reader, New, Me, Notifications)
* Pay attention at the loader dot the first time each section loads, and see if there are glitches
* Check that notifications work (orange bubble appearing and disappearing)
* Check that all the above work on mobile.
* If you have a single-site account, test everything works on that too.
* Check all the browsers we support (Chrome, Safari, Firefox, Edge, IE), mobile ones too (Safari, Chrome)!

Please take extra care in testing this one since it's a very important piece of the navigation.

---

#### Original Text:

Old:
![image](https://cloud.githubusercontent.com/assets/548849/11126157/e5af0d34-896d-11e5-8007-9f63ec5c22fd.png)

New (24px):
![image](https://cloud.githubusercontent.com/assets/548849/11126058/75ff12f4-896d-11e5-95b5-acfc07c17b41.png)
